packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -234,10 +258,34 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
     _focusNode = FocusNode();
   }
 
+  void expandParents(T parent) {
+    if (parent == null) return;
+
+    if (parent.parent?.index != -1) {
+      expandParents(parent.parent);
+    }
+
+    if (parent != null && !parent.isExpanded) {
+      _toggleNode(parent);
+    }
+  }
+
   @override
   void didUpdateWidget(TreeTable oldWidget) {
     super.didUpdateWidget(oldWidget);
 
+    cancel();
```

Comment:
you should only update if the oldWidget is different from the new widget.

Reply:
Done.